### PR TITLE
fix(gossipsub): fix mesh/fanout inconsistencies

### PR DESCRIPTION
**Pull request by jxs**
## Description

When a peer unsubscribes also remove it from fanout.

### CODE REVIEW

File: protocols/gossipsub/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
 ## 0.48.0
 
+- Correct state inconsistencies with the mesh and fanout when unsubscribing.
+  See [PR XXXX](https://github.com/libp2p/rust-libp2p/pull/XXXX)
```
Comment (dariusc93) [inline protocols/gossipsub/CHANGELOG.md]:
```suggestion
  See [PR 5690](https://github.com/libp2p/rust-libp2p/pull/5690)
```

File: protocols/gossipsub/CHANGELOG.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
   See [PR 5674](https://github.com/libp2p/rust-libp2p/pull/5674).
+
 - Apply `max_transmit_size` to the inner message instead of the final payload.
   See [PR 5642](https://github.com/libp2p/rust-libp2p/pull/5642).
 
 - Deprecate `void` crate.
   See [PR 5676](https://github.com/libp2p/rust-libp2p/pull/5676).
 
-## 0.47.1
-
```
Comment (elenaf9) [inline protocols/gossipsub/CHANGELOG.md:12]: Accidentally removed?

**Review by elenaf9 — APPROVED** [review]
LGTM

File: protocols/gossipsub/CHANGELOG.md (hunk 2 above)
Comment (dariusc93) [reply protocols/gossipsub/CHANGELOG.md:12]: I believe its due to the breaking changes from previous PRs that it was decided to skip 0.47.1 to 0.48.0.

Comment (jxs) [reply protocols/gossipsub/CHANGELOG.md:12]: yeah exactly, thanks Darius!

**Pull request merged**